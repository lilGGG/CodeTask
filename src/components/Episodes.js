import React,  { useState, useEffect } from 'react';
import ReactTable from "react-table-6";
import "react-table-6/react-table.css";
import {getEpisodes} from "../api/DataService.js";
import styled from "styled-components";


const EpisodesTitle = styled.h1`
  font-size: 1em;
  text-align: left;
  color: chocolate;
`;
const ListEpisodes = styled.ul`
  list-style-type: none;
`;

function Episodes(props) {

    const [episodes, setEpisodes] = useState({});
    // const [episodes1, setEpisodes1] = useState({});
    //  console.log("seasons componenti meej", props.seasons);
     
    //  let seasonSelected = (id) => {
    //      console.log("id",id);
    //     // var lowerCase = e.target.value.toLowerCase();
    //     // setSearchText(lowerCase);
    //   };

    useEffect(() => {
        async function fetchDataEpisodes() {
          let epis_result = await getEpisodes(props.seasonId);
          setEpisodes(epis_result);
        }
        fetchDataEpisodes();
        // setVisible(false);
      }, [props.seasonId]);
    
      console.log("episodes", episodes);

      const data = Array.from(episodes);

    //   const episFunc = (episodes) => {
    //     setEpisodes1(episodes);
    //   };

     
      
    return (
        <> 
         <ListEpisodes>
            <EpisodesTitle> Episodes </EpisodesTitle> 
           {data.map((epis) => (
                <> 
                <li> {epis.image? <a href={epis.url} target="_blank">  <img  src={epis.image.medium}  /> </a>: ""} </li>
                <li>  {epis.name}  </li>
                </>
            ))}
        </ListEpisodes>
        </>
    )
}

export default Episodes