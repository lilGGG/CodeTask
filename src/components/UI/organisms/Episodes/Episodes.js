import React,  { useState, useEffect } from 'react';
import ReactTable from "react-table-6";
import "react-table-6/react-table.css";
import {getEpisodes} from "../../../../api/DataService.js";
import styled from "styled-components";
import  EpisSeasonItem from '../../molecules/EpisSeasonItem/EpisSeasonItem';

const EpisodesTitle = styled.h1`
  font-size: 1em;
  text-align: left;
  color: chocolate;
`;
const ListEpisodes = styled.ul`
  list-style-type: none;
`;

function Episodes(props) {
    const [episodes, setEpisodes] = useState({});

    useEffect(() => {
        async function fetchDataEpisodes() {
          let epis_result = await getEpisodes(props.seasonId);
          setEpisodes(epis_result);
        }
        fetchDataEpisodes();
      }, [props.seasonId]);
    
      console.log("episodes", episodes);
      const data = Array.from(episodes);

    return (
        <> 
         <ListEpisodes>
            <EpisodesTitle> Episodes </EpisodesTitle> 
           {data.map((epis) => (
                <> 
                 <EpisSeasonItem
                   id ={epis.id}
                   number ={epis.number}
                   itemName={epis.name}
                   srcImg = {epis.image? epis.image.medium: ""}
                   img = {epis.image}
                   url={epis.url}
                />
                </>
            ))}
        </ListEpisodes>
        </>
    )
}

export default Episodes