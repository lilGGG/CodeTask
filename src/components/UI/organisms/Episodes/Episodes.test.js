import React from "react";
import Episodes from "./Episodes";
import ReactDom from "react-dom";
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('Episodes test', () => {
    it('renders snapshot', () => {
      const tree = renderer.create(<Episodes episodes  />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
  
