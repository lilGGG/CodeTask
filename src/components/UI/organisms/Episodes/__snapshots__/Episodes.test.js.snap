// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes test renders snapshot 1`] = `
<ul
  className="sc-hKMtZM gRNQnS"
>
  <h1
    className="sc-dkzDqf lmEGEz"
  >
     Episodes 
  </h1>
</ul>
`;
